backtester/engine/live_test.go

```diff
@@ -65,21 +64,33 @@ func TestSetupLiveDataHandler(t *testing.T) {
 
 func TestStart(t *testing.T) {
 	t.Parallel()
-	dc := &dataChecker{
-		shutdown: make(chan bool),
-	}
-	err := dc.Start()
-	assert.NoError(t, err)
+	t.Run("successful start", func(t *testing.T) {
+		t.Parallel()
+		dc := &dataChecker{
+			shutdown: make(chan bool),
+		}
+		err := dc.Start()
+		assert.NoError(t, err)
+
+		close(dc.shutdown)
+		dc.wg.Wait()
+	})
 
-	close(dc.shutdown)
-	dc.wg.Wait()
-	atomic.CompareAndSwapUint32(&dc.started, 0, 1)
-	err = dc.Start()
-	assert.ErrorIs(t, err, engine.ErrSubSystemAlreadyStarted)
+	t.Run("already started", func(t *testing.T) {
+		t.Parallel()
+		dc := &dataChecker{
+			started: 1,
+		}
+		err := dc.Start()
+		assert.ErrorIs(t, err, engine.ErrSubSystemAlreadyStarted)
+	})
```

Comment:
🐒 Optional nitpick:
This doesn't feel like it needs to be in a t.Run at all.
And if it's not, we could just re-use the `dc` from above.
Can also fold the `err` into the `assert` line, since it's so simple.